### PR TITLE
Data-driven Keyboard Conversions: C

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Converts `rules.mk` entries to data-driven where applicable. Renames `info.json` to `keyboard.json` in instances where `rules.mk` can be safely deleted.

- keyboards with names beginning with `c`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
## CI Build Failure:

### Failed: `checkerboards/snop60:via`

```
Copying checkerboards_snop60_via.hex to qmk_firmware folder                                         [OK]
Checking file size of checkerboards_snop60_via.hex
 * The firmware is too large! 28988/28672 (316 bytes over)
```

**Comment by fauxpark** [comment]
Looks like that board previously had `ENCODER_ENBALE = yes` instead of `ENABLE`.

File: keyboards/checkerboards/snop60/keyboard.json [hunk 1]
```diff
@@ -47,6 +47,16 @@
     },
     "processor": "atmega32u4",
     "bootloader": "atmel-dfu",
+    "features": {
```
Comment (zvecr) [inline keyboards/checkerboards/snop60/keyboard.json:50]: Could mitigate the oversize issue?
```suggestion
    "build": {
        "lto": true
    }
    "features": {
```

File: keyboards/coarse/vinta/info.json
Comment (zvecr) [inline keyboards/coarse/vinta/info.json:1]: Should this file be renamed to `keyboard.json` given the removal of `rules.mk`?

**Review by fauxpark — APPROVED** [review]

**Pull request merged**